test/test_get.py

```diff
@@ -196,37 +210,147 @@ def test_download_zip_and_expiration():
     downloader.cache_dict["test_resource"]["date_downloaded"] = str(
         datetime.now() - timedelta(days=8)
     )
+    downloader.cache_dict["test_api"]["date_downloaded"] = str(
+        datetime.now() - timedelta(days=8)
+    )
 
-    paths = downloader.download(resource)
+    paths = downloader.download(resource, api)
     # should download again
     assert paths[0] is not None
+    assert paths[1] is not None
+
+
+@pytest.mark.parametrize(
+    "downloader",
+    [
+        "downloader_without_specified_cache_dir",
+        "downloader_with_specified_cache_dir",
+    ],
+    indirect=True,
+)
+def test_cache_api_request(downloader):
+    api1 = APIRequest(
+        name="uniprot_api",
+        url_s=[
+            "https://rest.uniprot.org/uniprotkb/P12345.json",
+            "https://rest.uniprot.org/uniprotkb/P69905.json",
+        ],
+        lifetime=1,
+    )
+    api2 = APIRequest(
+        name="intact_api",
+        url_s="https://www.ebi.ac.uk/intact/ws/interactor/countTotal",
+        lifetime=1,
+    )
+
+    paths = downloader.download(api1, api2)
+
+    # test download list
+    assert isinstance(paths, list)
+    assert len(paths) == 3  # 2 API requests from api1, 1 API request from api2
+    for path in paths:
+        assert os.path.exists(path)
+    assert f"{os.sep}{api1.name}{os.sep}P12345.json" in paths[0]
+    assert f"{os.sep}{api1.name}{os.sep}P69905.json" in paths[1]
+    assert f"{os.sep}{api2.name}{os.sep}countTotal.json" in paths[2]
+
+    # api1 and api2 have been cached
+    # test cached api request
+    test_paths = downloader.download(
+        api1, api2
+    )  # get the path(s) of cached API request(s)
+    assert isinstance(test_paths, list)
+    assert len(paths) == len(test_paths)
+
+    paths.sort()
+    test_paths.sort()
+    for i in range(len(paths)):
+        with open(paths[i], "r") as file1:
+            api_request1 = json.load(file1)
+
+        with open(test_paths[i], "r") as file2:
+            api_request2 = json.load(file2)
+        assert api_request1 == api_request2
+
+
+@pytest.mark.parametrize(
+    "downloader",
+    [
+        "downloader_without_specified_cache_dir",
+        "downloader_with_specified_cache_dir",
+    ],
+    indirect=True,
+)
+def test_path_splitted(downloader):
+    resourc1 = Resource(
+        "test_resource1",
+        "https://github.com/biocypher/biocypher/raw/main/biocypher/_config/test_config.yaml",
+        lifetime=7,
+    )
+    resource2 = Resource(
+        name="test_resource2",
+        url_s=[
+            "https://github.com/biocypher/biocypher/raw/main/biocypher/_config/test_config.yaml",
+            "https://github.com/biocypher/biocypher/raw/main/biocypher/_config/test_schema_config_disconnected.yaml",
+        ],
+    )
+
+    paths = downloader.download(resourc1, resource2)
+    assert isinstance(paths, list)
+    assert len(paths) == 3  # if len(paths) == 3, then not splitted
+
+
+def test_download_with_parameter():
```

Comment:
Looks like indeed the download works now. However, looking at the returned path, the parameter (`?download=1`) is included in the file name, which I am not sure we want. Should we remove parameters from downloaded files? When downloading manually, we would just receive a `CollecTRI_source.tsv`.